Koalas GeoSeries union bug

**Describe the bug**
Koalas GeoSeries union got exception while postgis return normal result.

**To Reproduce**

Koalas GeoSeries test code:

```python
>>> from arctern_spark.geoseries import GeoSeries
>>> col1=['MULTIPOINT EMPTY']
>>> col2=['MULTIPOINT EMPTY']
>>> geo1=GeoSeries(col1)
>>> geo2=GeoSeries(col2)
>>> geo1.union(geo2)
20/07/13 16:20:40 ERROR Executor: Exception in task 5.0 in stage 2.0 (TID 24)12]
java.lang.IllegalArgumentException: Empty Points cannot be represented in WKB
```

postgis test sql:

```sql
postgres=# select st_union('MULTIPOINT EMPTY'::geometry, 'MULTIPOINT EMPTY'::geometry);
      st_union      
--------------------
 010400000000000000
(1 row)

postgres=# select st_astext(st_union('MULTIPOINT EMPTY'::geometry, 'MULTIPOINT EMPTY'::geometry));
    st_astext     
------------------
 MULTIPOINT EMPTY
(1 row)

```

**Any logs, error output, snapshot, etc?**

log was below:

```
20/07/13 16:20:40 ERROR Executor: Exception in task 5.0 in stage 2.0 (TID 24)12]
java.lang.IllegalArgumentException: Empty Points cannot be represented in WKB
        at org.locationtech.jts.io.WKBWriter.writePoint(WKBWriter.java:334)
        at org.locationtech.jts.io.WKBWriter.write(WKBWriter.java:308)
        at org.locationtech.jts.io.WKBWriter.write(WKBWriter.java:290)
        at org.apache.spark.sql.arctern.GeometryUDT$.ToWkb(GeometryUDT.scala:47)
        at org.apache.spark.sql.arctern.GeometryUDT$.GeomSerialize(GeometryUDT.scala:38)
        at org.apache.spark.sql.arctern.GeometryUDT.GeomSerialize(GeometryUDT.scala)
        at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage5.processNext(Unknown Source)
        at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
        at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:729)
        at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
        at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:132)
        at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
        at org.apache.spark.scheduler.Task.run(Task.scala:127)
        at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:444)
        at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1377)
        at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:447)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
20/07/13 16:20:40 WARN TaskSetManager: Lost task 5.0 in stage 2.0 (TID 24, spark-slave, executor driver): java.lang.IllegalArgumentException: Empty Points cannot be represented in WKB
        at org.locationtech.jts.io.WKBWriter.writePoint(WKBWriter.java:334)
        at org.locationtech.jts.io.WKBWriter.write(WKBWriter.java:308)
        at org.locationtech.jts.io.WKBWriter.write(WKBWriter.java:290)
        at org.apache.spark.sql.arctern.GeometryUDT$.ToWkb(GeometryUDT.scala:47)
        at org.apache.spark.sql.arctern.GeometryUDT$.GeomSerialize(GeometryUDT.scala:38)
        at org.apache.spark.sql.arctern.GeometryUDT.GeomSerialize(GeometryUDT.scala)
        at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage5.processNext(Unknown Source)
        at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
        at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:729)
        at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
        at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:132)
        at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
        at org.apache.spark.scheduler.Task.run(Task.scala:127)
        at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:444)
        at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1377)
        at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:447)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

20/07/13 16:20:40 ERROR TaskSetManager: Task 5 in stage 2.0 failed 1 times; aborting job
20/07/13 16:20:40 WARN TaskSetManager: Lost task 23.0 in stage 2.0 (TID 47, spark-slave, executor driver): TaskKilled (Stage cancelled)
20/07/13 16:20:40 WARN TaskSetManager: Lost task 25.0 in stage 2.0 (TID 49, spark-slave, executor driver): TaskKilled (Stage cancelled)
20/07/13 16:20:40 WARN TaskSetManager: Lost task 27.0 in stage 2.0 (TID 51, spark-slave, executor driver): TaskKilled (Stage cancelled)
20/07/13 16:20:40 WARN TaskSetManager: Lost task 24.0 in stage 2.0 (TID 48, spark-slave, executor driver): TaskKilled (Stage cancelled)
20/07/13 16:20:40 WARN TaskSetManager: Lost task 26.0 in stage 2.0 (TID 50, spark-slave, executor driver): TaskKilled (Stage cancelled)
20/07/13 16:20:40 WARN TaskSetManager: Lost task 28.0 in stage 2.0 (TID 52, spark-slave, executor driver): TaskKilled (Stage cancelled)
20/07/13 16:20:40 WARN TaskSetManager: Lost task 29.0 in stage 2.0 (TID 53, spark-slave, executor driver): TaskKilled (Stage cancelled)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ljq/miniconda3/envs/koalas_regression/lib/python3.7/site-packages/arctern_spark-0.0.0-py3.7.egg/arctern_spark/geoseries.py", line 282, in __repr__
    pser = wkt_ks._to_internal_pandas()
  File "/home/ljq/miniconda3/envs/koalas_regression/lib/python3.7/site-packages/databricks/koalas/series.py", line 5156, in _to_internal_pandas
    return first_series(self._internal.to_pandas_frame)
  File "/home/ljq/miniconda3/envs/koalas_regression/lib/python3.7/site-packages/databricks/koalas/utils.py", line 465, in wrapped_lazy_property
    setattr(self, attr_name, fn(self))
  File "/home/ljq/miniconda3/envs/koalas_regression/lib/python3.7/site-packages/databricks/koalas/internal.py", line 755, in to_pandas_frame
    pdf = sdf.toPandas()
  File "/home/ljq/miniconda3/envs/koalas_regression/lib/python3.7/site-packages/pyspark-3.0.0-py3.7.egg/pyspark/sql/pandas/conversion.py", line 138, in toPandas
    pdf = pd.DataFrame.from_records(self.collect(), columns=self.columns)
  File "/home/ljq/miniconda3/envs/koalas_regression/lib/python3.7/site-packages/pyspark-3.0.0-py3.7.egg/pyspark/sql/dataframe.py", line 596, in collect
    sock_info = self._jdf.collectToPython()
  File "/home/ljq/miniconda3/envs/koalas_regression/lib/python3.7/site-packages/py4j-0.10.9-py3.7.egg/py4j/java_gateway.py", line 1305, in __call__
    answer, self.gateway_client, self.target_id, self.name)
  File "/home/ljq/miniconda3/envs/koalas_regression/lib/python3.7/site-packages/pyspark-3.0.0-py3.7.egg/pyspark/sql/utils.py", line 131, in deco
    return f(*a, **kw)
  File "/home/ljq/miniconda3/envs/koalas_regression/lib/python3.7/site-packages/py4j-0.10.9-py3.7.egg/py4j/protocol.py", line 328, in get_return_value
    format(target_id, ".", name), value)
py4j.protocol.Py4JJavaError: An error occurred while calling o360.collectToPython.
: org.apache.spark.SparkException: Job aborted due to stage failure: Task 5 in stage 2.0 failed 1 times, most recent failure: Lost task 5.0 in stage 2.0 (TID 24, spark-slave, executor driver): java.lang.IllegalArgumentException: Empty Points cannot be represented in WKB
        at org.locationtech.jts.io.WKBWriter.writePoint(WKBWriter.java:334)
        at org.locationtech.jts.io.WKBWriter.write(WKBWriter.java:308)
        at org.locationtech.jts.io.WKBWriter.write(WKBWriter.java:290)
        at org.apache.spark.sql.arctern.GeometryUDT$.ToWkb(GeometryUDT.scala:47)
        at org.apache.spark.sql.arctern.GeometryUDT$.GeomSerialize(GeometryUDT.scala:38)
        at org.apache.spark.sql.arctern.GeometryUDT.GeomSerialize(GeometryUDT.scala)
        at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage5.processNext(Unknown Source)
        at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
        at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:729)
        at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
        at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:132)
        at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
        at org.apache.spark.scheduler.Task.run(Task.scala:127)
        at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:444)
        at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1377)
        at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:447)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

Driver stacktrace:
        at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2023)
        at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:1972)
        at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:1971)
        at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
        at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
        at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
        at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:1971)
        at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:950)
        at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:950)
        at scala.Option.foreach(Option.scala:407)
        at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:950)
        at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2203)
        at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2152)
        at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2141)
        at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
        at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:752)
        at org.apache.spark.SparkContext.runJob(SparkContext.scala:2093)
        at org.apache.spark.SparkContext.runJob(SparkContext.scala:2114)
        at org.apache.spark.SparkContext.runJob(SparkContext.scala:2133)
        at org.apache.spark.SparkContext.runJob(SparkContext.scala:2158)
        at org.apache.spark.rdd.RDD.$anonfun$collect$1(RDD.scala:1004)
        at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
        at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
        at org.apache.spark.rdd.RDD.withScope(RDD.scala:388)
        at org.apache.spark.rdd.RDD.collect(RDD.scala:1003)
        at org.apache.spark.sql.execution.SparkPlan.executeCollect(SparkPlan.scala:385)
        at org.apache.spark.sql.Dataset.$anonfun$collectToPython$1(Dataset.scala:3448)
        at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:3616)
        at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$5(SQLExecution.scala:100)
        at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:160)
        at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:87)
        at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:763)
        at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)
        at org.apache.spark.sql.Dataset.withAction(Dataset.scala:3614)
        at org.apache.spark.sql.Dataset.collectToPython(Dataset.scala:3445)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
        at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
        at py4j.Gateway.invoke(Gateway.java:282)
        at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
        at py4j.commands.CallCommand.execute(CallCommand.java:79)
        at py4j.GatewayConnection.run(GatewayConnection.java:238)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalArgumentException: Empty Points cannot be represented in WKB
        at org.locationtech.jts.io.WKBWriter.writePoint(WKBWriter.java:334)
        at org.locationtech.jts.io.WKBWriter.write(WKBWriter.java:308)
        at org.locationtech.jts.io.WKBWriter.write(WKBWriter.java:290)
        at org.apache.spark.sql.arctern.GeometryUDT$.ToWkb(GeometryUDT.scala:47)
        at org.apache.spark.sql.arctern.GeometryUDT$.GeomSerialize(GeometryUDT.scala:38)
        at org.apache.spark.sql.arctern.GeometryUDT.GeomSerialize(GeometryUDT.scala)
        at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage5.processNext(Unknown Source)
        at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
        at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:729)
        at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
        at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:132)
        at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
        at org.apache.spark.scheduler.Task.run(Task.scala:127)
        at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:444)
        at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1377)
        at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:447)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        ... 1 more
```

--- a/scala/src/main/scala/org/apache/spark/sql/arctern/CodeGenUtil.scala
+++ b/scala/src/main/scala/org/apache/spark/sql/arctern/CodeGenUtil.scala
@@ -87,7 +87,7 @@ object CodeGenUtil {
         s"""
            |${mutableGeometryInitCode(geoName)}
            |${geoName} = $callFunc;
-           |${emptyGeometryCheck(geoName)}
+           |${emptyPointCheck(geoName)}
            |$value = ${serialGeometryCode(geoName)}
            |""".stripMargin
       case _ => s"$value = $callFunc;"
@@ -112,5 +112,5 @@ object CodeGenUtil {
 
   def utf8StringFromStringCode(string_name: String) = s"org.apache.spark.unsafe.types.UTF8String.fromString($string_name);"
 
-  def emptyGeometryCheck(geoName: String) = s"if ($geoName.isEmpty()) $geoName = new org.locationtech.jts.geom.GeometryFactory().createGeometryCollection();"
+  def emptyPointCheck(geoName: String) = s"""if ($geoName.getGeometryType() == "Point" && $geoName.isEmpty()) $geoName = new org.locationtech.jts.geom.GeometryFactory().createGeometryCollection();"""
 }

--- a/scala/src/main/scala/org/apache/spark/sql/arctern/CodeGenUtil.scala
+++ b/scala/src/main/scala/org/apache/spark/sql/arctern/CodeGenUtil.scala
@@ -87,6 +87,7 @@ object CodeGenUtil {
         s"""
            |${mutableGeometryInitCode(geoName)}
            |${geoName} = $callFunc;
+           |${emptyGeometryCheck(geoName)}
            |$value = ${serialGeometryCode(geoName)}
            |""".stripMargin
       case _ => s"$value = $callFunc;"
@@ -110,4 +111,6 @@ object CodeGenUtil {
   }
 
   def utf8StringFromStringCode(string_name: String) = s"org.apache.spark.unsafe.types.UTF8String.fromString($string_name);"
+
+  def emptyGeometryCheck(geoName: String) = s"if ($geoName.isEmpty()) $geoName = new org.locationtech.jts.geom.GeometryFactory().createGeometryCollection();"
 }

--- a/scala/src/main/scala/org/apache/spark/sql/arctern/expressions/Functions.scala
+++ b/scala/src/main/scala/org/apache/spark/sql/arctern/expressions/Functions.scala
@@ -334,7 +334,7 @@ case class ST_Centroid(inputsExpr: Seq[Expression]) extends ST_UnaryOp {
 
   override def expr: Expression = inputsExpr(0)
 
-  override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = codeGenJob(ctx, ev, geo => s"$geo.getCentroid().isEmpty() ? new org.locationtech.jts.geom.GeometryFactory().createGeometryCollection() : $geo.getCentroid()")
+  override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = codeGenJob(ctx, ev, geo => s"$geo.getCentroid()")
 
   override def dataType: DataType = new GeometryUDT
 
@@ -432,7 +432,7 @@ case class ST_Intersection(inputsExpr: Seq[Expression]) extends ST_BinaryOp {
 
   override def rightExpr: Expression = inputsExpr(1)
 
-  override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = codeGenJob(ctx, ev, (left, right) => s"$left.intersection($right).isEmpty() ? new org.locationtech.jts.geom.GeometryFactory().createGeometryCollection() : $left.intersection($right)")
+  override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = codeGenJob(ctx, ev, (left, right) => s"$left.intersection($right)")
 
   override def dataType: DataType = new GeometryUDT
 
@@ -693,7 +693,7 @@ case class ST_SymDifference(inputsExpr: Seq[Expression]) extends ST_BinaryOp {
 
   override def rightExpr: Expression = inputsExpr(1)
 
-  override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = codeGenJob(ctx, ev, (left, right) => s"$left.symDifference($right).isEmpty() ? new org.locationtech.jts.geom.GeometryFactory().createGeometryCollection() : $left.symDifference($right)")
+  override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = codeGenJob(ctx, ev, (left, right) => s"$left.symDifference($right)")
 
   override def dataType: DataType = new GeometryUDT
 
@@ -706,7 +706,7 @@ case class ST_Difference(inputsExpr: Seq[Expression]) extends ST_BinaryOp {
 
   override def rightExpr: Expression = inputsExpr(1)
 
-  override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = codeGenJob(ctx, ev, (left, right) => s"$left.difference($right).isEmpty() ? new org.locationtech.jts.geom.GeometryFactory().createGeometryCollection() : $left.difference($right)")
+  override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = codeGenJob(ctx, ev, (left, right) => s"$left.difference($right)")
 
   override def dataType: DataType = new GeometryUDT
 

--- a/scala/src/test/scala/FunctionsTest.scala
+++ b/scala/src/test/scala/FunctionsTest.scala
@@ -2284,6 +2284,7 @@ class FunctionsTest extends AdapterTest {
       Row(GeometryUDT.FromWkt("MULTIPOINT (0 0, 1 0, 1 2, 1 2)"), GeometryUDT.FromWkt("MULTILINESTRING ( (0 0, 1 2), (0 0, 1 0, 1 1),(-1 2,3 4,1 -3,-2 1) )")),
       Row(GeometryUDT.FromWkt("MULTILINESTRING ( (0 0, 1 2), (0 0, 1 0, 1 1),(-1 2,3 4,1 -3,-2 1) )"), GeometryUDT.FromWkt("MULTIPOLYGON ( ((0 0, 1 4, 1 0,0 0)) )")),
       Row(GeometryUDT.FromWkt("MULTIPOLYGON ( ((0 0, 1 4, 1 0,0 0)) )"), GeometryUDT.FromWkt("POINT (1 5)")),
+      Row(GeometryUDT.FromWkt("MULTIPOINT EMPTY"), GeometryUDT.FromWkt("MULTIPOINT EMPTY")),
     )
 
     val schema = StructType(Array(StructField("left_geo", new GeometryUDT, nullable = true), StructField("right_geo", new GeometryUDT, nullable = true)))
@@ -2304,6 +2305,7 @@ class FunctionsTest extends AdapterTest {
     assert(collect(4).getAs[GeometryUDT](0).toString == "MULTILINESTRING ((0 0, 1 2), (0 0, 1 0, 1 1), (-1 2, 3 4, 1 -3, -2 1))")
     assert(collect(5).getAs[GeometryUDT](0).toString == "GEOMETRYCOLLECTION (LINESTRING (-1 2, 0.7142857142857143 2.857142857142857), LINESTRING (1 3, 3 4, 1 -3, -2 1), POLYGON ((1 0, 0 0, 0.7142857142857143 2.857142857142857, 1 4, 1 3, 1 2, 1 1, 1 0)))")
     assert(collect(6).getAs[GeometryUDT](0).toString == "GEOMETRYCOLLECTION (POINT (1 5), POLYGON ((0 0, 1 4, 1 0, 0 0)))")
+    assert(collect(7).getAs[GeometryUDT](0).toString == "GEOMETRYCOLLECTION EMPTY")
 
     val rst2 = df.select(st_union(col("left_geo"), col("right_geo")))
     rst2.show(false)
@@ -2317,6 +2319,7 @@ class FunctionsTest extends AdapterTest {
     assert(collect2(4).getAs[GeometryUDT](0).toString == "MULTILINESTRING ((0 0, 1 2), (0 0, 1 0, 1 1), (-1 2, 3 4, 1 -3, -2 1))")
     assert(collect2(5).getAs[GeometryUDT](0).toString == "GEOMETRYCOLLECTION (LINESTRING (-1 2, 0.7142857142857143 2.857142857142857), LINESTRING (1 3, 3 4, 1 -3, -2 1), POLYGON ((1 0, 0 0, 0.7142857142857143 2.857142857142857, 1 4, 1 3, 1 2, 1 1, 1 0)))")
     assert(collect2(6).getAs[GeometryUDT](0).toString == "GEOMETRYCOLLECTION (POINT (1 5), POLYGON ((0 0, 1 4, 1 0, 0 0)))")
+    assert(collect2(7).getAs[GeometryUDT](0).toString == "GEOMETRYCOLLECTION EMPTY")
   }
 
   test("ST_Union-Null") {

